Source/Csla/Configuration/Fluent/SerializationOptions.cs

```diff
@@ -7,44 +7,41 @@
 //-----------------------------------------------------------------------
 
 using Csla.Serialization;
-using Csla.Serialization.Mobile;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace Csla.Configuration
 {
   /// <summary>
   /// Use this type to configure the settings for serialization.
   /// </summary>
-  public class SerializationOptions
+  /// <param name="cslaOptions">The CSLA options.</param>
+  public class SerializationOptions(CslaOptions cslaOptions)
   {
+    /// <summary>
+    /// Gets the CSLA .NET configuration options.
+    /// </summary>
+    protected CslaOptions CslaOptions => cslaOptions;
+
     /// <summary>
     /// Sets the serialization formatter type used by CSLA .NET
     /// for all explicit object serialization (such as cloning,
     /// n-level undo, etc). Default is MobileFormatter.
     /// </summary>
-    public SerializationOptions SerializationFormatter<T>() where T: ISerializationFormatter
+    public SerializationOptions UseSerializationFormatter<T>() where T : ISerializationFormatter
     {
-      ApplicationContext.SerializationFormatter = typeof(T);
+      CslaOptions.Services.TryAddTransient(typeof(ISerializationFormatter), typeof(T));
       return this;
     }
 
     /// <summary>
-    /// Sets type of the writer that is used to write data to 
-    /// serialization stream in SerializationFormatterFactory.GetFormatter().
+    /// Gets the serialization formatter type used by CSLA .NET.
     /// </summary>
-    public SerializationOptions MobileWriter<T>() where T : ICslaWriter
-    {
-      CslaReaderWriterFactory.WriterType = typeof(T);
-      return this;
-    }
+    public Type SerializationFormatterType 
```

Comment:
For me this is still awkward. I would just add `MobileFormatter` as a default into the DI with `AddTransient<ISerializationFormatter, MobileFormatter>`  and the user config adds his implementation after that one. Default behvior of MS DI (and all others as far as I know) the last registered implementation is the default one. If you want _all_ you have to request an `IEnumerable` of the interface.